### PR TITLE
feat(contrib/linode): add Linode provision scripts and docs

### PR DESCRIPTION
**Pull request by croemmich**
Linode's recent switch to KVM gave me an added push to rewrite our team's provisioning tools to work with the new hypervisor as it makes running CoreOS much simpler. This time, I did so with the intention of fully documenting the process for the community and our team at Myriad Mobile as well as make our tools publicly available. So here's a stab at that!

### CODE REVIEW

**Comment by carmstrong** [comment]
This is awesome, @croemmich!! We'll give it a review!

File: contrib/linode/linode-user-data-template.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    command: start
+    content: |
+      [Unit]
+      Description=Disable IPv6. The notion of a security group does not exist on Linode. To make firewalling each Linode easier, disable IPv6.
+      Before=network.target
+
+      [Service]
+      Type=oneshot
+      ExecStart=/bin/sh -c "sysctl -w net.ipv6.conf.all.disable_ipv6=1"
+    # use ntpd rather than timesyncd. timesyncd was not keeping the nodes close enough to keep ceph happy
```
Comment (carmstrong) [inline contrib/linode/linode-user-data-template.yaml]: Perhaps we should file this issue upstream. I'm sure the CoreOS team would want to know that this change had this effect... the time sync daemon was changed in CoreOS 681: https://coreos.com/releases/#681.0.0

Comment (croemmich) [inline contrib/linode/linode-user-data-template.yaml]: Agreed, I've submitted https://github.com/coreos/bugs/issues/391. Unfortunately systemd-timesyncd doesn't seem heavily documented or mature :disappointed:

File: docs/installing_deis/linode.rst [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+.. code-block:: console
+
+    $ ssh-keygen -q -t rsa -f ~/.ssh/deis -N '' -C deis
+
+
+Check System Requirements
+-------------------------
+
+Please refer to :ref:`system-requirements` for resource considerations when choosing a Linode
+plan to run Deis. The smallest recommended cluster consists of 3 Linode 4096s.
```
Comment (mboersma) [inline docs/installing_deis/linode.rst]: Could you replace with or add this wording? We just added this to the other "Installing" docs for consistency:
```
A Deis cluster must have 3 or more nodes. See :ref:`cluster-size` for more details.
```

**Comment by mboersma** [comment]
This is fantastic. The python could use some comments but is well-crafted and easy to follow. Code LGTM.

**Comment by bacongobbler** [comment]
I used to have a Ramnode account, but no Linode account. Sorry!

**Comment by croemmich** [comment]
@mboersma I've updated the docs to include your wording. Additionally I have added validation to the provision script to ensure the specified Linode plan and node number is consistent with the system requirements.

**Comment by carmstrong** [comment]
I think this is fine to :ship: since @croemmich mentioned in the docs that it's supported by our community. Great work! Code LGTM.

**Comment by mboersma** [comment]
This passed the only relevant tests: the docs. Nothing under `/contrib` is tested by CI. Merging...

**Pull request merged**